addon/props/model_props.py

```diff
@@ -86,6 +86,21 @@ class SOURCEOPS_ModelProps(bpy.types.PropertyGroup):
         default=False,
     )
 
+    transform_source: bpy.props.EnumProperty(
+        name='Transform Source',
+        description='Method of specifying $origin and $scale.\nEither manually specified in this panel, or via an object',
+        items=[
+            ('MANUAL', 'Manual Input', 'Specify the Transform manually in this panel'),
+            ('OBJECT', 'Object', 'Use an Object\'s transforms\nIf it is None, then falls back to Manual Input')
+        ]
+    )
+
+    transform_object: bpy.props.PointerProperty(
+        name='Transform Object',
+        description='The object to use the transform of as the $origin, $scale and Z-Axis rotation',
```

Comment:
I'd go with `transforms` (plural) in the description.
No need to mention Z-Axis rotation, because $origin includes it, and you don't mention it in the earlier description (see my first comment in this review).